aiosqlite/core.py

```diff
@@ -116,6 +118,19 @@ def set_exception(fut, e):
 
                 get_loop(future).call_soon_threadsafe(set_exception, future, e)
 
+        # Clean up within this thread only if the parent event loop exits ungracefully
+        if not self._running or self._connection is None or self._parent_loop is None:
```

Comment:
This isn't obvious as to when it will return or run the cleanup below. Is there a way to always try cleaning up?